Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -289,5 +292,18 @@ public void RegisterForDisposal(IDisposable disposable)
         public void Dispose() => _disposables.Dispose();
 
         string IKernel.Name => Name;
+
+        public void SetProperty<T>(T property) where T : class
+        {
+            if (!_propertyBag.TryAdd(typeof(T), property))
+            {
+                throw new InvalidOperationException($"Cannot add property with key {typeof(T)}.");
+            }
+        }
+
+        public T GetProperty<T>() where T : class
```

Comment:
This should throw if the property is not found.

Reply:
ok, makes sense